extensions/procfs/CpuStat.h

```diff
@@ -34,7 +34,7 @@ class CpuStatData {
  public:
   static std::optional<CpuStatData> parseCpuStatLine(std::istream& iss);
 
-  auto operator<=>(const CpuStatData& rhs) const = default;
+  bool operator>(const CpuStatData& rhs) const;
```

Comment:
What was the problem with the default spaceship operator?

Reply:
Clang complained that the compared types are not default comparable so the default cooperation operator was implicitly deleted.

Reply:
I'm not sure if we can, the compiler complains about not having a viable three-way comparison function for the members, which are of `SystemClockDuration` aka `std::chrono::duration<uint64_t, std::ratio<1, HZ>>` type. If that's not comparable according to clang I don't think we can do much about it.

Reply:
There is an implementation in TimeUtil.h, try including that

Reply:
Thanks for pointing this out, it seems that this implementation was a workaround for this exact issue, but it was restricted to versions under libc++ 16. Removing this restrictions fixes the issue in 359181bf99d0e3b8cac9e0db92edd2c10eef42f5